content/02.body.md

```diff
@@ -92,26 +92,32 @@ When permuting bipartite networks, our method ensures that each node's class mem
 Similarly, heterogeneous networks should be permuted by considering each edge type as a separate network [@doi:10.1371/journal.pcbi.1004259; @doi:10.15363/thinklab.d136].
 This way, each node retains its within-edge-type degree for all edge types.
 We provide documentation for parameter choices depending on the type of network being permuted in the GitHub repository (<https://github.com/hetio/xswap>).
-The original algorithm and our proposed modification are given in Figure {@fig:algo}.
+The original algorithm and our proposed modification are given in Figures {@fig:algo} and {@fig:algodigram}.
 
 ![
   **XSwap algorithm pseudocode.**
   **A.** XSwap algorithm presented by Hanhijärvi, et al. [@doi:10.1137/1.9781611972795.67].
   **B.** Extension of the XSwap algorithm to other types of networks.
 ](images/algorithms_label.png){#fig:algo width="65%"}
 
+![
+  **Modified XSwap algorithm graphical explanation.**
+](images/XSwap.svg){#fig:algodiagram width="65%"}
```

Comment:
Nice. I wonder if this is actually a bipartite graph instead of an undirected. It could technically be either, but never is there an edge within sides.

Reply:
Definitely. Would it be more clear if I added some additional edges on the left or right side to make it non-bipartite?

Reply:
Yes an additional edge or a layout that didn't appear to just be two columns would fix this. Leave it to you whether that's feasible given the re-submission timelime

Reply:
Just added it!